src/app/icd/ICDConfigurationData.h

```diff
@@ -25,6 +25,11 @@ namespace chip {
 namespace app {
 // Forward declaration of ICDManager to allow it to be friend with ICDConfigurationData
 class ICDManager;
+
+// Forward declaration of TestICDManager to allow it to be friend with the ICDConfigurationData
+// Used in unit tests
+class TestICDManager;
```

Comment:
Does it actually need to be friend? Could a public method have done the job rather than allowing reaching down the throat?

Reply:
For now, yes. We don't want to expose a public API that changes the durations without going trough the ICDManager since it is responsible of all the buisness logic. 

Since the ICDManager doesn't yet support changing the durations, we allow the test to have direct access to the private function like the ICDManager. This will most likely be phased out once the change can be done from the ICDManager.